src/Application/Services/OpcAlarmMonitoredItemService.cs

```diff
@@ -0,0 +1,178 @@
+﻿// SPDX-License-Identifier: MIT. 
+// Copyright Contributors to the Open Manufacturing Platform.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OMP.Connector.Application.Extensions;
+using OMP.Connector.Application.OpcUa;
+using OMP.Connector.Domain;
+using OMP.Connector.Domain.Configuration;
+using OMP.Connector.Domain.Extensions;
+using OMP.Connector.Domain.Models.Telemetry;
+using OMP.Connector.Domain.OpcUa;
+using OMP.Connector.Domain.OpcUa.Services;
+using OMP.Connector.Domain.Schema;
+using OMP.Connector.Domain.Schema.Alarms;
+using Opc.Ua;
+using Opc.Ua.Client;
+
+namespace OMP.Connector.Application.Services
+{
+    public class OpcAlarmMonitoredItemService : MonitoredItem, IOpcAlarmMonitoredItemService
+    {
+        private bool _disposedValue;
+        private readonly ILogger _logger;
+        private readonly IMapper _mapper;
+        private readonly IMessageSender _messageSender;
+        private readonly ConnectorConfiguration _connectorConfiguration;
+        private IComplexTypeSystem _complexTypeSystem;
+        private OpcUaMonitoredItem _monitoredItemCommand;
+
+        public TelemetryMessageMetadata MessageMetadata { get; set; }
+
+        public OpcAlarmMonitoredItemService(
+            IOptions<ConnectorConfiguration> connectorConfiguration,
+            IMessageSender messageSender,
+            IMapper mapper,
+            ILogger<OpcAlarmMonitoredItemService> logger)
+        {
+            this._logger = logger;
+            this._mapper = mapper;
+            this._connectorConfiguration = connectorConfiguration.Value;
+            this._messageSender = messageSender;
+
+            this.Notification += this.OnNotification;
+        }
+
+        public void Initialize(AlarmSubscriptionMonitoredItem monitoredItemCommand, IComplexTypeSystem complexTypeSystem, TelemetryMessageMetadata messageMetadata, Session session)
```

Comment:
Is it possible to use our abstraction IOpcUaSession object instead of this concrete Session object?

Reply:
Done